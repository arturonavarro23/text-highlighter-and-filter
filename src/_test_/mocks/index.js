
export const defaultState = {
  text: 'Test words',
  colors: ['red', 'yellow', 'green'],
  selectedColor: 'red',
  selectedFilters: ['red', 'yellow'],
  ranges: []
};
