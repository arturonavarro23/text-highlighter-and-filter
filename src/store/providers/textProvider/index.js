import React, { createContext, useContext, useReducer } from 'react';

export const TextContext = createContext();

export const TextProvider = ({ reducer, initialState, children }) => (
  <TextContext.Provider value={useReducer(reducer, initialState)}>
    {children}
  </TextContext.Provider>
);

export const useStateValue = () => useContext(TextContext);
