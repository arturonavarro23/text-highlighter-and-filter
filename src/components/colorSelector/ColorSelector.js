import React from 'react';
import { useStateValue } from '../../store/providers/textProvider';
import { setColor } from '../../store/actions';
import Button from '../common/button';

const ColorSelector = () => {
  const [{ colors, selectedColor }, dispatch] = useStateValue();

  const onColorButtonClick = color => () => {
    dispatch(setColor(selectedColor === color ? null : color));
  }

  const renderButton = color => (
    <Button
      key={color}
      variant={color}
      onClick={onColorButtonClick(color)}
      selected={color === selectedColor}
    />
  );

  return (
    <React.Fragment>
      <h1>
        Colors
      </h1>
      {colors.map(renderButton)}
    </React.Fragment>
  );
}

export default ColorSelector;
