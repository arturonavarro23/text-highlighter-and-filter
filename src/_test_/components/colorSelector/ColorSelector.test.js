import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import * as TextProvider from '../../../store/providers/textProvider';
import ColorSelector from '../../../components/colorSelector';
import { defaultState } from '../../mocks';

describe('<ColorSelector> test', () => {
  it('Snapshot should render correctly', () => {
    jest
      .spyOn(TextProvider, 'useStateValue')
      .mockImplementation(() => [defaultState]);

    const colorSelectorTree = renderer
      .create(<ColorSelector />)
      .toJSON();

    expect(colorSelectorTree).toMatchSnapshot();
  });
  it('Should render all buttons', () => {
    jest
      .spyOn(TextProvider, 'useStateValue')
      .mockImplementation(() => [defaultState]);

    const colorSelector = mount(<ColorSelector />);
    expect(colorSelector.find('button')).toHaveLength(defaultState.colors.length);
  });
  it('Should have selected button', () => {
    jest
      .spyOn(TextProvider, 'useStateValue')
      .mockImplementation(() => [defaultState]);

    const colorSelector = mount(<ColorSelector />);
    const selectedButton = colorSelector.find('.selected');
    expect(selectedButton).toHaveLength(1);
    expect(selectedButton.hasClass(`btn--${defaultState.selectedColor}`)).toBe(true);
  });
  it('onColorButtonClick should be called', () => {

    const mockFn = jest.fn().mockName('dispatch');

    jest
      .spyOn(TextProvider, 'useStateValue')
      .mockImplementation(() => [defaultState, mockFn]);

    const colorSelector = mount(<ColorSelector />);
    const selectedButton = colorSelector.find('.selected');
    selectedButton.simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});
