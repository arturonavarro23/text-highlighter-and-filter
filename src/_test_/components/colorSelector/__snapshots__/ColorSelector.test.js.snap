// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ColorSelector> test Snapshot should render correctly 1`] = `
Array [
  <h1>
    Colors
  </h1>,
  <button
    className="btn btn--red selected"
    onClick={[Function]}
  />,
  <button
    className="btn btn--yellow"
    onClick={[Function]}
  />,
  <button
    className="btn btn--green"
    onClick={[Function]}
  />,
]
`;
